phoenix-core/src/main/java/org/apache/phoenix/schema/types/PDataType.java

```diff
@@ -1144,17 +1144,18 @@ public static PDataType fromLiteral(Object value) {
       return null;
     }
     for (PDataType type : PDataType.values()) {
-      if (type.isArrayType()) {
-        PhoenixArray arr = (PhoenixArray) value;
-        if ((type.getSqlType() == arr.baseType.sqlType + PDataType.ARRAY_TYPE_BASE)
-            && type.getJavaClass().isInstance(value)) {
-          return type;
-        }
-      } else {
-        if (type.getJavaClass().isInstance(value)) {
-          return type;
-        }
-      }
+			if (type.isArrayType() && type.getJavaClass().isInstance(value)) {
```

Comment:
I will fix the formatting error here. But it is required to check the isInstance() in first IF itself otherwise it throw ClassCastException. For any value whose data type ordinal is less than PhoenixArray will try to be type casted to PhoenixArray at 1148 which is wrong as it is iterated in the order of ordinal where PhoenixArray type will be iterated before PJsonDataType.